presto-hive/src/main/java/com/facebook/presto/hive/HiveClientModule.java

```diff
@@ -139,18 +143,25 @@ public ExecutorService createCachingHiveMetastoreExecutor(HiveConnectorId hiveCl
     @Provides
     public HiveCluster createHiveCluster(Injector injector, HivePluginConfig config)
     {
-        URI uri = checkNotNull(config.getMetastoreUri(), "metastoreUri is null");
-        String scheme = uri.getScheme();
-        checkArgument(!isNullOrEmpty(scheme), "metastoreUri scheme is missing: %s", uri);
+        List<URI> uris = checkNotNull(config.getMetastoreUri(), "metastoreUri is null");
+        checkArgument(uris.size() > 0, "metastoreUri must specify at least one URI");
+        String scheme = uris.get(0).getScheme();
+        checkArgument(!isNullOrEmpty(scheme), "metastoreUri scheme is missing: %s", uris.get(0));
         switch (scheme) {
             case "discovery":
                 return injector.getInstance(DiscoveryLocatedHiveCluster.class);
             case "thrift":
-                checkArgument(uri.getHost() != null, "metastoreUri host is missing: %s", uri);
-                checkArgument(uri.getPort() != -1, "metastoreUri port is missing: %s", uri);
-                HostAndPort address = HostAndPort.fromParts(uri.getHost(), uri.getPort());
-                return new StaticHiveCluster(address, injector.getInstance(HiveMetastoreClientFactory.class));
+                List<HostAndPort> addresses = transform(uris, HiveClientModule::getHostAndPort);
+                return new StaticHiveCluster(addresses, injector.getInstance(HiveMetastoreClientFactory.class));
         }
-        throw new IllegalArgumentException("unsupported metastoreUri: " + uri);
+        throw new IllegalArgumentException("unsupported metastoreUris: " + Arrays.toString(uris.toArray()));
```

Comment:
This conversion isn't necessary, since a collection should provide a usable `toString()`